pkg/cloud/vsphere/vsphere_cloud_config/config_test.go

```diff
@@ -0,0 +1,182 @@
+    datacenters:
+    - DC0
+    - DC1
+labels:
+  zone: k8s-zone
+  region: k8s-region
+`
+
+const multiVCDCsConfigINI = `
+[Global]
+port = 443
+insecure-flag = true
+secret-name = "global-secret"
+secret-namespace = "global-secret-ns"
+
+[VirtualCenter "t1"]
+server = "10.0.0.1"
+datacenters = "DC0,DC1,DC2"
+secret-name = "tenant1-secret"
+secret-namespace = "kube-system"
+
+[VirtualCenter "10.0.0.2"]
+datacenters = "DC3"
+
+[VirtualCenter "10.0.0.3"]
+datacenters = "DC5,DC6"
+ip-family = "ipv6"
+`
+
+const invalidConfig = `boom[]{}`
+
+const invalidConfigWrongGlobalPort = `
+[Global]
+port = -443
+insecure-flag = true
+`
+
+const invalidConfigWrongVCPort = `
+[Global]
+port = 443
+insecure-flag = true
+
+[VirtualCenter "10.0.0.3"]
+datacenters = "DC5,DC6"
+port = -1
+ip-family = "ipv6"
+`
+
+func TestINIConfigConversion(t *testing.T) {
+
+	t.Run("basic config yaml conversion", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+		configStruct, err := ReadConfig([]byte(basicConfigINI))
+		g.Expect(err).Should(gmg.Succeed())
+
+		convertedConfig, err := MarshalConfig(configStruct)
+		g.Expect(err).Should(gmg.Succeed())
+		// Trim left emptyline just for keep constant more readable
+		g.Expect(convertedConfig).To(gmg.BeEquivalentTo(strings.TrimLeft(basicConfigYaml, "\n")))
+	})
+
+	t.Run("basic config yaml conversion with vc section", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+		configStruct, err := ReadConfig([]byte(basicConfigVcenterSectionINI))
+		g.Expect(err).Should(gmg.Succeed())
+
+		convertedConfig, err := MarshalConfig(configStruct)
+		g.Expect(err).Should(gmg.Succeed())
+		// Trim left emptyline just for keep constant more readable
+		g.Expect(convertedConfig).To(gmg.BeEquivalentTo(strings.TrimLeft(basicConfigVcenterSectionYAML, "\n")))
+	})
+
+	t.Run("Test multi DCs config yaml conversion", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+		configStruct, err := ReadConfig([]byte(multiVCDCsConfigINI))
+		g.Expect(err).Should(gmg.Succeed())
+
+		g.Expect(len(configStruct.Vcenter)).To(gmg.Equal(3))
+
+		VC1 := configStruct.Vcenter["t1"]
+		g.Expect(len(VC1.Datacenters)).To(gmg.Equal(3))
+		g.Expect(VC1.Datacenters).To(gmg.BeComparableTo([]string{"DC0", "DC1", "DC2"}))
+		g.Expect(VC1.VCenterIP).To(gmg.Equal("10.0.0.1"))
+		g.Expect(VC1.SecretNamespace).To(gmg.Equal("kube-system"))
+		g.Expect(VC1.SecretName).To(gmg.Equal("tenant1-secret"))
+
+		VC2 := configStruct.Vcenter["10.0.0.2"]
+		g.Expect(len(VC2.Datacenters)).To(gmg.Equal(1))
+		g.Expect(VC2.Datacenters).To(gmg.BeComparableTo([]string{"DC3"}))
+		g.Expect(VC2.VCenterIP).To(gmg.Equal("10.0.0.2"))
+		g.Expect(VC2.IPFamilyPriority).To(gmg.Equal([]string{}))
+
+		VC3 := configStruct.Vcenter["10.0.0.3"]
+		g.Expect(len(VC3.Datacenters)).To(gmg.Equal(2))
+		g.Expect(VC3.Datacenters).To(gmg.BeComparableTo([]string{"DC5", "DC6"}))
+		g.Expect(VC3.IPFamilyPriority).To(gmg.Equal([]string{"ipv6"}))
+
+		_, err = MarshalConfig(configStruct)
+		g.Expect(err).Should(gmg.Succeed())
```

Comment:
Not checking the expected output of marshal config on this one?

Reply:
Nope, my intention was to ensure that the config struct will be marshaled without errors (sort of smoke test).

As I understand ordering would not be guaranteed for `vcenter` section and if I will compare it with some sort of expected string it might be flaky. 
At the moment of writing, I thought that just checking errors should be enough and extra effort for checking output here is not worth it.

Reply:
Please change this error line to "to not have occurred"